relay-profiling/src/sample/v2.rs

```diff
@@ -0,0 +1,159 @@
+use std::collections::{BTreeMap, HashSet};
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::ProfileError;
+use crate::measurements::Measurement;
+use crate::sample::{DebugMeta, Frame, ThreadMetadata, Version};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ProfileMetadata {
+    /// Random UUID identifying a chunk
+    chunk_id: String,
+    /// Random UUID for each profiler session
+    profiler_id: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    debug_meta: Option<DebugMeta>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    environment: Option<String>,
+    platform: String,
+    release: String,
+
+    /// Hard-coded string containing "2" to indicate the format version.
+    version: Version,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Sample {
+    /// Unix timestamp in seconds with millisecond precision when the sample
+    /// was captured.
+    timestamp: f64,
+    stack_id: usize,
+    thread_id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ProfileChunk {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    measurements: Option<BTreeMap<String, Measurement>>,
```

Comment:
Do we need a semantic difference between `None` and an empty map? If not, I would go with `#[serde(default, skip_serializing_if = "BTreeMap::is_empty")]`.